src/cpp/src/llm_pipeline_static.hpp

```diff
@@ -9,23 +9,42 @@
 
 namespace ov {
 namespace genai {
+namespace static_llm {
 
 struct ModelConfigDesc {
     std::string type;
     std::string name_or_path;
     int num_key_value_heads;
 };
 
-class StaticLLMPipeline final : public LLMPipelineImplBase {
+struct LLMPipelineFactory {
+    static std::unique_ptr<LLMPipelineImplBase> create(const std::filesystem::path& path,
+                                                       const ov::genai::Tokenizer& tokenizer,
+                                                       const std::string& device,
+                                                       const ov::AnyMap& config);
+
+    static std::unique_ptr<LLMPipelineImplBase> create(const std::filesystem::path& path,
+                                                       const std::string& device,
+                                                       const ov::AnyMap& config);
+
+    static std::unique_ptr<LLMPipelineImplBase> create(const std::shared_ptr<ov::Model>& model,
+                                                       const ModelConfigDesc& model_desc,
+                                                       const ov::genai::Tokenizer& tokenizer,
+                                                       const std::string& device,
+                                                       const ov::AnyMap& properties,
+                                                       const ov::genai::GenerationConfig& generation_config = {});
+};
+
+class StatefulLLMPipeline : public LLMPipelineImplBase {
 public:
-    StaticLLMPipeline(
+    StatefulLLMPipeline(
         const std::filesystem::path& path,
         const ov::genai::Tokenizer& tokenizer,
         const std::string& device,
         const ov::AnyMap& config
     );
 
-    StaticLLMPipeline(
+    StatefulLLMPipeline(
```

Comment:
it's weird we have `static_llm::StatefulLLMPipeline`.

Maybe it's time we name it:
`npu_impl::StaticLLMPipeline` - old dual model pipeline
`npu_impl::StatefulLLMPipeline` - the new one with dynamic shapes and states

and whatever else...
`npu_impl::SingleModelLLMPipeline`
`npu_impl::StaticWhisperPipeline`


CC: @dmatveev